utils/torch_solver.py

```diff
@@ -11,13 +13,20 @@
 class TorchSolver(BaseSolver):
     """Torch base solver"""
 
-    stopping_strategy = 'callback'
+    stopping_criterion = SufficientProgressCriterion(
+        patience=20, strategy='callback'
+    )
 
-    def set_objective(self, pl_module, trainer, tf_model, tf_dataset):
+    parameters = {
+        'batch_size': [64],
+    }
+
+    def set_objective(self, pl_module, torch_dataset, tf_model, tf_dataset):
         self.pl_module = pl_module
-        self.main_trainer = trainer  # we use this in order
-        # to access some elements from the trainer when
-        # initializing it below
+        self.dataloader = torch.utils.data.DataLoader(
+            torch_dataset, batch_size=self.batch_size,
+            num_workers=6
```

Comment:
we should add a todo here to replace this hard coded value with something more adaptable.

Reply:
Yes I agree. The `AUTOTUNE` fo `keras` is very powerfull for this and explains that the iterations are mostly faster there....